src/sample/courseRegController.java

```diff
@@ -48,67 +55,169 @@ public void ContactUs(MouseEvent event) throws IOException {
   public void openBrowser() throws URISyntaxException, IOException {
     openLink();
   }
-
-  public void addcourse() {
-
-    courseList.getItems().add("CAT201 - INTEGRATED SOFTWARE DEVELOPMENT WORKSHOP");
-    courseList.getItems().add("CMT221 - DATABASE ORGANISATIONS AND DESIGN");
-    courseList.getItems().add("CST232 - OPERATING SYSTEMS");
-    courseList.getItems().add("CSE241 - FOUNDATION OF SOFTWARE ENGINEERING");
-    courseList.getItems().add("CPT111 - PRINCIPLES OF PROGRAMMING");
-    courseList.getItems().add("CPT112 - DISCRETE STRUCTURES");
-    courseList.getItems().add("CST131 - COMPUTER ORGANISATIONS");
-    courseList.getItems().add("CPT113 - PROGRAMMING METHODOLOGY AND DATA STRUCTURES");
-    courseList.getItems().add("CPT115 - MATHEMATICAL METHODS FOR COMPUTER SCIENCE");
-    courseList.getItems().add("CPC151 - FUNDAMENTALS OF LOGIC AND ARTIFICIAL INTELLIGENCE");
-    courseList.getItems().add("AKW103 - INTRODUCTION TO MANAGEMENT");
-    courseList.getItems().add("AKW104 - ACCOUNTING AND FINANCE");
-    courseList.getItems().add("ATW202 - BUSINESS RESEARCH METHOD");
-    courseList.getItems().add("ATW241 - PRINCIPLES OF MARKETING");
-    courseList.getItems().add("ATW262 - PRINCIPLES OF FINANCE");
-    courseList.getItems().add("MAA101 - CALCULUS FOR SCIENCE STUDENTS");
-    courseList.getItems().add("MAT111 - LINEAR ALGEBRA");
-    courseList.getItems().add("MAT181 - PROGRAMMING FOR SCIENTIFIC APPLICATIONS");
-    courseList.getItems().add("MAA111 - ALGEBRA FOR SCIENCE STUDENTS");
-    courseList.getItems().add("LAK100 - KOREAN LANGUAGE");
-    courseList.getItems().add("LAJ100 - JAPANESE LANGUAGE");
-    courseList.getItems().add("LKM400 - BAHASA MALAYSIA");
-    courseList.getItems().add("LSP300 - ACADEMIC ENGLISH");
-    courseList.getItems().add("HFF225 - PHILOSOPHY AND CURRENT ISSUES");
-    courseList.getItems().add("HFE224 - APPRECIATION OF ETHICS AND CIVILISATIONS");
-    }
   
+  //Read and display list of all courses that can be registered from txt file
+  private void displaycourselist(){
+    try {
+      File fileObj = new File("student1courseList.txt");
+      Scanner fileReader = new Scanner(fileObj);
+      while (fileReader.hasNextLine()) {
+        courseList.getItems().add(fileReader.nextLine());
+      }
+      fileReader.close();
+    
+    } catch (FileNotFoundException e) {
+      System.out.println("An error occurred.");
```

Comment:
Did you output the "An error occured" to the users?

Reply:
No, this one only output in terminal. If want to output to the users, then need to provide a space in the app to output this error.